## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#718a69ec-c3f2-443c-8305-f5a226024292) |
| mint-gateway-s3.sh | :x: [more...](#3ae48397-e6e6-43e6-b3f6-e116ce68910f) |
| mint-gateway-azure.sh | :x: [more...](#7b30e3ed-01d5-403c-b366-7eaa2796d3ff) |
| mint-gateway-nas.sh | :x: [more...](#f0d342f5-45c7-4630-85a7-8bf89e728be5) |
<a name="f0d342f5-45c7-4630-85a7-8bf89e728be5"></a>
### 9526-574f91f/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:32481
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 76822018786c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 1 seconds
(5/15) Running awscli tests ... done in 1 minutes and 13 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 25 seconds
(8/15) Running minio-dotnet tests ... done in 27 seconds
(9/15) Running minio-go tests ... FAILED in 48 seconds
{}
(9/15) Running minio-java tests ... FAILED in 27 seconds
{
  "name": "minio-java",
  "function": "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
  "duration": 30,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidRequest, message = Bucket is missing ObjectLockConfiguration, bucketName = minio-java-test-2bbnjnh, objectName = minio-java-test-39469qe, resource = /minio-java-test-2bbnjnh/minio-java-test-39469qe, requestId = 160C4EBF628F5068, hostId = 95ab8957-ecf7-4bf1-a9ee-b4975084e8d2)\nrequest={method=PUT, url=http://minio-dev3.minio.io:32481/minio-java-test-2bbnjnh/minio-java-test-39469qe?legal-hold=, headers=Host: minio-dev3.minio.io:32481\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: erOisiAmIIzsh/Ze2QqGXw==\nx-amz-content-sha256: 96b73c95a8d33e664ab2170e095025b47ebd55978bb71cebd6a51e394bf96722\nx-amz-date: 20200506T021552Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200506/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 384\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.GOGET\nVary: Origin\nX-Amz-Request-Id: 160C4EBF628F5068\nX-Xss-Protection: 1; mode=block\nDate: Wed, 06 May 2020 02:15:52 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.executePut(MinioClient.java:1424), io.minio.MinioClient.enableObjectLegalHold(MinioClient.java:3568), FunctionalTest.enableObjectLegalHold_test(FunctionalTest.java:3303), FunctionalTest.runTests(FunctionalTest.java:4079), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(9/15) Running minio-js tests ... done in 38 seconds
(10/15) Running minio-py tests ... done in 56 seconds
(11/15) Running s3cmd tests ... done in 15 seconds
(12/15) Running s3select tests ... done in 1 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="7b30e3ed-01d5-403c-b366-7eaa2796d3ff"></a>
### 9526-574f91f/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:30840
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7f23c3a40cba:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 10 minutes and 12 seconds
(4/15) Running aws-sdk-ruby tests ... FAILED in 4 minutes and 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 242638.53,
  "error": "Net::ReadTimeout",
  "status": "FAIL"
}
(4/15) Running awscli tests ... FAILED in 50 seconds
{
  "name": "awscli",
  "duration": 3923,
  "function": "aws --endpoint-url http://minio-dev6.minio.io:30840 s3api copy-object --bucket awscli-mint-test-bucket-32583 --storage-class REDUCED_REDUNDANCY --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-32583/datafile-1-kB\n",
  "status": "FAIL",
  "error": "An error occurred (NotImplemented) when calling the CopyObject operation: A header you provided implies functionality that is not implemented"
}
(4/15) Running healthcheck tests ... done in 0 seconds
(5/15) Running mc tests ... done in 3 minutes and 7 seconds
(6/15) Running minio-dotnet tests ... done in 3 minutes and 7 seconds
(7/15) Running minio-go tests ... FAILED in 3 minutes and 59 seconds
{}
(7/15) Running minio-java tests ... FAILED in 28 minutes and 40 seconds
{
  "name": "minio-java",
  "function": "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
  "duration": 1797,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidRequest, message = Bucket is missing ObjectLockConfiguration, bucketName = minio-java-test-guvr2d, objectName = minio-java-test-21i85h6, resource = /minio-java-test-guvr2d/minio-java-test-21i85h6, requestId = 160C517AC43DCF7F, hostId = 7e394e30-ce90-41e0-be9d-9f26c9b71299)\nrequest={method=PUT, url=http://minio-dev6.minio.io:30840/minio-java-test-guvr2d/minio-java-test-21i85h6?legal-hold=, headers=Host: minio-dev6.minio.io:30840\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: erOisiAmIIzsh/Ze2QqGXw==\nx-amz-content-sha256: 96b73c95a8d33e664ab2170e095025b47ebd55978bb71cebd6a51e394bf96722\nx-amz-date: 20200506T030556Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200506/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 382\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.GOGET\nVary: Origin\nX-Amz-Request-Id: 160C517AC43DCF7F\nX-Xss-Protection: 1; mode=block\nDate: Wed, 06 May 2020 03:05:56 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.executePut(MinioClient.java:1424), io.minio.MinioClient.enableObjectLegalHold(MinioClient.java:3568), FunctionalTest.enableObjectLegalHold_test(FunctionalTest.java:3303), FunctionalTest.runTests(FunctionalTest.java:4079), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(7/15) Running minio-js tests ... FAILED in 7 minutes and 56 seconds
{
  "name": "minio-js",
  "function": "listObjectsV2WithMetadata(bucketName, prefix, recursive, startAfter)",
  "args": "bucketName:minio-js-test-7d848c0e-3eb8-4a02-8590-4d7897654ad6, recursive:true",
  "duration": 209,
  "status": "FAIL",
  "error": "TypeError: Cannot read property '0' of undefined at /mint/run/core/minio-js/node_modules/minio/dist/main/xml-parsers.js:414:42 at Array.forEach (<anonymous>) at parseListObjectsV2WithMetadata (node_modules/minio/dist/main/xml-parsers.js:409:21) at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:19) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(7/15) Running minio-py tests ... done in 12 minutes and 37 seconds
(8/15) Running s3cmd tests ... FAILED in 43 seconds
{
  "name": "s3cmd",
  "duration": "1633",
  "function": "test_copy_object",
  "status": "FAIL",
  "error": ""
}
(8/15) Running s3select tests ... done in 22 seconds
(9/15) Running security tests ... done in 0 seconds

Executed 9 out of 15 tests successfully.
```
<a name="3ae48397-e6e6-43e6-b3f6-e116ce68910f"></a>
### 9526-574f91f/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31957
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 48fce1ddaae8:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 10 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 26 seconds
(8/15) Running minio-dotnet tests ... done in 27 seconds
(9/15) Running minio-go tests ... FAILED in 46 seconds
{}
(9/15) Running minio-java tests ... FAILED in 30 seconds
{
  "name": "minio-java",
  "function": "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
  "duration": 46,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidRequest, message = Bucket is missing ObjectLockConfiguration, bucketName = minio-java-test-1oakmj0, objectName = minio-java-test-2vn4lhu, resource = /minio-java-test-1oakmj0/minio-java-test-2vn4lhu, requestId = 160C4EBF9FA33749, hostId = 3bd59e0f-1fd0-4992-b79c-df8d831b29f6)\nrequest={method=PUT, url=http://minio-c3.minio.io:31957/minio-java-test-1oakmj0/minio-java-test-2vn4lhu?legal-hold=, headers=Host: minio-c3.minio.io:31957\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: erOisiAmIIzsh/Ze2QqGXw==\nx-amz-content-sha256: 96b73c95a8d33e664ab2170e095025b47ebd55978bb71cebd6a51e394bf96722\nx-amz-date: 20200506T021553Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200506/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 384\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.GOGET\nVary: Origin\nX-Amz-Request-Id: 160C4EBF9FA33749\nX-Xss-Protection: 1; mode=block\nDate: Wed, 06 May 2020 02:15:53 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.executePut(MinioClient.java:1424), io.minio.MinioClient.enableObjectLegalHold(MinioClient.java:3568), FunctionalTest.enableObjectLegalHold_test(FunctionalTest.java:3303), FunctionalTest.runTests(FunctionalTest.java:4079), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(9/15) Running minio-js tests ... done in 37 seconds
(10/15) Running minio-py tests ... done in 58 seconds
(11/15) Running s3cmd tests ... done in 1 minutes and 0 seconds
(12/15) Running s3select tests ... done in 1 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="718a69ec-c3f2-443c-8305-f5a226024292"></a>
### 9526-574f91f/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30201
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 695f530a3031:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 3 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 45 seconds
(4/15) Running aws-sdk-ruby tests ... FAILED in 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2142.58,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(4/15) Running awscli tests ... FAILED in 12 seconds
{
  "name": "awscli",
  "duration": 7930,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "fatal error: An error occurred (XMinioServerNotInitialized) when calling the ListObjectsV2 operation (reached max retries: 4): Server not initialized, please try again.remove_bucket failed: Unable to delete all objects in the bucket, bucket will not be deleted."
}
(4/15) Running healthcheck tests ... done in 0 seconds
(5/15) Running mc tests ... FAILED in 8 seconds
{
  "name": "mc",
  "duration": "2960",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-19480 --quiet --no-color rm mc-test-object-4295.downloaded myminio/mc-test-bucket-29052/mc-test-object-4295  >>> \nRemoving `mc-test-object-4295.downloaded`.\nmc: <ERROR> Unable to list folder. Server not initialized, please try again.\nmc: <ERROR> Failed to remove `myminio/mc-test-bucket-29052/mc-test-object-4295`. "
}
(5/15) Running minio-dotnet tests ... FAILED in 17 seconds
{
  "name": "minio-dotnet : PresignedGetObject_Test2",
  "function": "Task<string> PresignedGetObjectAsync(string bucketName, string objectName, int expiresInt, Dictionary<string, string> reqParams = null)",
  "args": {
    "bucketName": "miniodotnet9dbhq3q9xov4puc",
    "objectName": "ad&{dd*b}]",
    "expiresInt": "0"
  },
  "duration": 109,
  "status": "FAIL",
  "alert": "",
  "message": "MinIO API responded with message=Server not initialized, please try again.",
  "error": "Server not initialized, please try again.: Minio.Exceptions.MinioException: MinIO API responded with message=Server not initialized, please try again.\n   at Minio.MinioClient.ParseError(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 394\n   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 540\n   at Minio.MinioClient.ExecuteTaskAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 383\n   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, String uploadId, Int32 partNumber, Byte[] data, Dictionary`2 metaData, Dictionary`2 sseHeaders, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 529\n   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, Stream data, Int64 size, String contentType, Dictionary`2 metaData, ServerSideEncryption sse, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 298\n   at Minio.Functional.Tests.FunctionalTest.PresignedGetObject_Test2(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2162"
}
(5/15) Running minio-go tests ... FAILED in 1 minutes and 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-qdst9eaxzxsbky99",
    "fileName": "",
    "objectName": "",
    "opts": ""
  },
  "duration": 61914,
  "error": "Server not initialized, please try again.",
  "function": "FPutObject(bucketName, objectName, fileName, opts)",
  "message": "MakeBucket failed",
  "name": "minio-go: testFPutObjectV2",
  "status": "FAIL"
}
(5/15) Running minio-java tests ... FAILED in 6 minutes and 55 seconds
{
  "name": "minio-java",
  "function": "presignedPutObject(String bucketName, String objectName, Integer expires)",
  "duration": 15,
  "status": "FAIL",
  "error": "java.lang.Exception: failed to create object. Response: Response{protocol=http/1.1, code=503, message=Service Unavailable, url=http://minio-dev7.minio.io:30201/minio-java-test-1b6chjp/minio-java-test-1d6s8rc?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20200506%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20200506T022117Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=15ba544894b1d77b050cf691cb20327163cdbab575e3a942c83ba98408a2fb95}, Response body: <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key>minio-java-test-1d6s8rc</Key><BucketName>minio-java-test-1b6chjp</BucketName><Resource>/minio-java-test-1b6chjp/minio-java-test-1d6s8rc</Resource><RequestId>160C4F0B03D7415D</RequestId><HostId>4cc73130-fdae-4145-bffd-750b3f01e467</HostId></Error> >>> [FunctionalTest.writeObject(FunctionalTest.java:233), FunctionalTest.presignedPutObject_test2(FunctionalTest.java:2073), FunctionalTest.runTests(FunctionalTest.java:4062), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(5/15) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(5/15) Running minio-py tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "function": "make_bucket(bucket_name, location)",
  "args": {
    "bucket_name": "minio-py-test-5c945db2-8441-4e11-8f86-a1e5737ef12d",
    "location": "default value ('us-east-1')"
  },
  "duration": 600556,
  "message": "HTTPConnectionPool(host='minio-dev7.minio.io', port=30201): Max retries exceeded with url: /minio-py-test-5c945db2-8441-4e11-8f86-a1e5737ef12d/ (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 188, in test_make_bucket_default_region\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 371, in make_bucket\n    headers=headers)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 336, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 805, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 439, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev7.minio.io', port=30201): Max retries exceeded with url: /minio-py-test-5c945db2-8441-4e11-8f86-a1e5737ef12d/ (Caused by ResponseError('too many 503 error responses',))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1968, in main\n    test_make_bucket_default_region(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 196, in test_make_bucket_default_region\n    raise Exception(err)\nException: HTTPConnectionPool(host='minio-dev7.minio.io', port=30201): Max retries exceeded with url: /minio-py-test-5c945db2-8441-4e11-8f86-a1e5737ef12d/ (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(5/15) Running s3cmd tests ... FAILED in 45 seconds
{
  "name": "s3cmd",
  "duration": "45452",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(5/15) Running s3select tests ... done in 6 seconds
(6/15) Running security tests ... done in 0 seconds

Executed 6 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/buildscripts/verify-build.sh
+++ b/buildscripts/verify-build.sh
@@ -154,34 +154,6 @@ function run_test_erasure_sets() {
     return "$rv"
 }
 
-function run_test_dist_erasure_sets_ipv6()
-{
-    minio_pids=( $(start_minio_dist_erasure_sets_ipv6) )
-
-    export SERVER_ENDPOINT="[::1]:9000"
-
-    (cd "$WORK_DIR" && "$FUNCTIONAL_TESTS")
-    rv=$?
-
-    for pid in "${minio_pids[@]}"; do
-        kill "$pid"
-    done
-    sleep 3
-
-    if [ "$rv" -ne 0 ]; then
-        for i in $(seq 0 9); do
-            echo "server$i log:"
-            cat "$WORK_DIR/dist-minio-v6-900$i.log"
-        done
-    fi
-
-    for i in $(seq 0 9); do
-        rm -f "$WORK_DIR/dist-minio-v6-900$i.log"
-    done
-
-    return "$rv"
-}
-
 function run_test_zone_erasure_sets()
 {
     minio_pids=( $(start_minio_zone_erasure_sets) )

--- a/cmd/server-main.go
+++ b/cmd/server-main.go
@@ -208,20 +208,24 @@ func initSafeMode() (err error) {
 	//    version is needed, migration is needed etc.
 	rquorum := InsufficientReadQuorum{}
 	wquorum := InsufficientWriteQuorum{}
-	optimeout := OperationTimedOut{}
-	for n := range newRetryTimerSimple(retryCtx) {
+	for range newRetryTimerSimple(retryCtx) {
 		// let one of the server acquire the lock, if not let them timeout.
 		// which shall be retried again by this loop.
-		if err = txnLk.GetLock(leaderLockTimeout); err == nil {
-			// Migrate all backend configs to encrypted backend configs, optionally
-			// handles rotating keys for encryption, if there is any retriable failure
-			// that shall be retried if there is an error.
-			if err = handleEncryptedConfigBackend(newObject, true); err == nil {
-				// Upon success migrating the config, initialize all sub-systems
-				// if all sub-systems initialized successfully return right away
-				if err = initAllSubsystems(newObject); err == nil {
-					return nil
-				}
+		if err = txnLk.GetLock(leaderLockTimeout); err != nil {
+			logger.Info("Waiting for all MinIO sub-systems to be initialized.. trying to acquire lock")
+			continue
+		}
+		logger.Info("Waiting for all MinIO sub-systems to be initialized.. lock acquired")
+
+		// Migrate all backend configs to encrypted backend configs, optionally
+		// handles rotating keys for encryption, if there is any retriable failure
+		// that shall be retried if there is an error.
+		if err = handleEncryptedConfigBackend(newObject, true); err == nil {
+			// Upon success migrating the config, initialize all sub-systems
+			// if all sub-systems initialized successfully return right away
+			if err = initAllSubsystems(newObject); err == nil {
+				// All successful return.
+				return nil
 			}
 		}
 
@@ -230,15 +234,11 @@ func initSafeMode() (err error) {
 			errors.Is(err, errConfigNotFound) ||
 			errors.Is(err, context.Canceled) ||
 			errors.Is(err, context.DeadlineExceeded) ||
-			errors.As(err, &optimeout) ||
 			errors.As(err, &rquorum) ||
 			errors.As(err, &wquorum) ||
 			isErrBucketNotFound(err) {
-			if n < 5 {
-				logger.Info("Waiting for all MinIO sub-systems to be initialized..")
-			} else {
-				logger.Info("Waiting for all MinIO sub-systems to be initialized.. possible cause (%v)", err)
-			}
+			logger.Info("Waiting for all MinIO sub-systems to be initialized.. possible cause (%v)", err)
+			txnLk.Unlock() // Unlock the transaction lock and allow other nodes to acquire the lock if possible.
 			continue
 		}
 
@@ -256,10 +256,41 @@ func initSafeMode() (err error) {
 }
 
 func initAllSubsystems(newObject ObjectLayer) (err error) {
-	// List buckets to be re-used for loading configs.
-	buckets, err := newObject.ListBuckets(GlobalContext)
-	if err != nil {
-		return fmt.Errorf("Unable to list buckets: %w", err)
+	// %w is used by all error returns here to make sure
+	// we wrap the underlying error, make sure when you
+	// are modifying this code that you do so, if and when
+	// you want to add extra context to your error. This
+	// ensures top level retry works accordingly.
+	var buckets []BucketInfo
+	if globalIsDistXL || globalIsXL {
+		// List buckets to heal, and be re-used for loading configs.
+		buckets, err = newObject.ListBucketsHeal(GlobalContext)
+		if err != nil {
+			return fmt.Errorf("Unable to list buckets to heal: %w", err)
+		}
+		// Attempt a heal if possible and re-use the bucket names
+		// to reload their config.
+		wquorum := &InsufficientWriteQuorum{}
+		rquorum := &InsufficientReadQuorum{}
+		for _, bucket := range buckets {
+			if err = newObject.MakeBucketWithLocation(GlobalContext, bucket.Name, ""); err != nil {
+				if errors.As(err, &wquorum) || errors.As(err, &rquorum) {
+					// Retrun the error upwards for the caller to retry.
+					return fmt.Errorf("Unable to heal bucket: %w", err)
+				}
+				if _, ok := err.(BucketExists); !ok {
+					// ignore any other error and log for investigation.
+					logger.LogIf(GlobalContext, err)
+					continue
+				}
+				// Bucket already exists, nothing that needs to be done.
+			}
+		}
+	} else {
+		buckets, err = newObject.ListBuckets(GlobalContext)
+		if err != nil {
+			return fmt.Errorf("Unable to list buckets: %w", err)
+		}
 	}
 
 	// Initialize config system.

--- a/cmd/xl-sets.go
+++ b/cmd/xl-sets.go
@@ -22,6 +22,7 @@ import (
 	"hash/crc32"
 	"io"
 	"net/http"
+	"sort"
 	"strings"
 	"sync"
 	"time"
@@ -1689,20 +1690,18 @@ func (s *xlSets) HealObject(ctx context.Context, bucket, object string, opts mad
 
 // Lists all buckets which need healing.
 func (s *xlSets) ListBucketsHeal(ctx context.Context) ([]BucketInfo, error) {
-	listBuckets := []BucketInfo{}
-	var healBuckets = map[string]BucketInfo{}
+	var listBuckets []BucketInfo
+	var healBuckets = make(map[string]VolInfo)
 	for _, set := range s.sets {
-		buckets, _, err := listAllBuckets(set.getDisks())
-		if err != nil {
+		// lists all unique buckets across drives.
+		if err := listAllBuckets(set.getDisks(), healBuckets); err != nil {
 			return nil, err
 		}
-		for _, currBucket := range buckets {
-			healBuckets[currBucket.Name] = BucketInfo(currBucket)
-		}
 	}
-	for _, bucketInfo := range healBuckets {
-		listBuckets = append(listBuckets, bucketInfo)
+	for _, v := range healBuckets {
+		listBuckets = append(listBuckets, BucketInfo(v))
 	}
+	sort.Sort(byBucketName(listBuckets))
 	return listBuckets, nil
 }
 

--- a/cmd/xl-v1-healing.go
+++ b/cmd/xl-v1-healing.go
@@ -173,11 +173,7 @@ func healBucket(ctx context.Context, storageDisks []StorageAPI, bucket string, w
 
 // listAllBuckets lists all buckets from all disks. It also
 // returns the occurrence of each buckets in all disks
-func listAllBuckets(storageDisks []StorageAPI) (buckets map[string]VolInfo,
-	bucketsOcc map[string]int, err error) {
-
-	buckets = make(map[string]VolInfo)
-	bucketsOcc = make(map[string]int)
+func listAllBuckets(storageDisks []StorageAPI, healBuckets map[string]VolInfo) (err error) {
 	for _, disk := range storageDisks {
 		if disk == nil {
 			continue
@@ -188,7 +184,7 @@ func listAllBuckets(storageDisks []StorageAPI) (buckets map[string]VolInfo,
 			if IsErrIgnored(err, bucketMetadataOpIgnoredErrs...) {
 				continue
 			}
-			return nil, nil, err
+			return err
 		}
 		for _, volInfo := range volsInfo {
 			// StorageAPI can send volume names which are
@@ -197,13 +193,14 @@ func listAllBuckets(storageDisks []StorageAPI) (buckets map[string]VolInfo,
 			if isReservedOrInvalidBucket(volInfo.Name, false) {
 				continue
 			}
-			// Increase counter per bucket name
-			bucketsOcc[volInfo.Name]++
-			// Save volume info under bucket name
-			buckets[volInfo.Name] = volInfo
+			// always save unique buckets across drives.
+			if _, ok := healBuckets[volInfo.Name]; !ok {
+				healBuckets[volInfo.Name] = volInfo
+			}
+
 		}
 	}
-	return buckets, bucketsOcc, nil
+	return nil
 }
 
 // Only heal on disks where we are sure that healing is needed. We can expand

--- a/cmd/xl-zones.go
+++ b/cmd/xl-zones.go
@@ -44,16 +44,6 @@ func (z *xlZones) SingleZone() bool {
 	return len(z.zones) == 1
 }
 
-func (z *xlZones) quickHealBuckets(ctx context.Context) {
-	bucketsInfo, err := z.ListBucketsHeal(ctx)
-	if err != nil {
-		return
-	}
-	for _, bucket := range bucketsInfo {
-		z.MakeBucketWithLocation(ctx, bucket.Name, "")
-	}
-}
-
 // Initialize new zone of erasure sets.
 func newXLZones(ctx context.Context, endpointZones EndpointZones) (ObjectLayer, error) {
 	var (
@@ -88,7 +78,6 @@ func newXLZones(ctx context.Context, endpointZones EndpointZones) (ObjectLayer, 
 		}
 	}
 
-	z.quickHealBuckets(ctx)
 	go intDataUpdateTracker.start(GlobalContext, localDrives...)
 	return z, nil
 }

